packages/fiori/src/NotificationList.ts

```diff
@@ -1,54 +1,169 @@
-import NavigationMode from "@ui5/webcomponents-base/dist/types/NavigationMode.js";
-import ListItemBase from "@ui5/webcomponents/dist/ListItemBase.js";
-import List from "@ui5/webcomponents/dist/List.js";
-import NotificationListGroupItem from "./NotificationListGroupItem.js";
+import type {
+	ListItemClickEventDetail,
+	ListItemToggleEventDetail,
+	ListItemCloseEventDetail,
+} from "@ui5/webcomponents/dist/List.js";
+import NotificationListItemBase from "./NotificationListItemBase.js";
+import NotificationListInternal from "./NotificationListInternal.js";
+
+// Template
+import NotificationListTemplate from "./generated/templates/NotificationListTemplate.lit.js";
 
 // Texts
 import {
 	NOTIFICATION_LIST_ACCESSIBLE_NAME,
 } from "./generated/i18n/i18n-defaults.js";
 
+type NotificationItemEventDetail = {
+	item: NotificationListItemBase,
+}
+
+type NotificationItemClickEventDetail = NotificationItemEventDetail;
+type NotificationItemToggleEventDetail = NotificationItemEventDetail;
+type NotificationItemCloseEventDetail = NotificationItemEventDetail;
+
 /**
  * @class
- *
+ * ### Overview
  * The `ui5-notification-list` web component represents
  * a container for `ui5-li-notification-group` and `ui5-li-notification`.
  *
+ * ### ES6 Module Import
+ *
+ * `import "@ui5/webcomponents/dist/NotificationList.js";``
  * @constructor
- * @extends List
+ * @extends UI5Element
  * @since 2.0
  * @public
  */
-@customElement("ui5-notification-list")
+@customElement({
+	tag: "ui5-notification-list",
+	renderer: litRender,
+	languageAware: true,
+	styles: [],
+	template: NotificationListTemplate,
+	dependencies: [
+		NotificationListInternal,
+	],
+})
+
+/**
+ * Fired when an item is clicked.
+ * @allowPreventDefault
+ * @param {HTMLElement} item The clicked item.
+ * @public
+ */
+@event<NotificationItemClickEventDetail>("item-click", {
+	detail: {
+		/**
+		 * @public
+		 */
+		item: { type: HTMLElement },
+	},
+})
 
-class NotificationList extends List {
-	navigationMode = NavigationMode.Auto;
+/**
+ * Fired when the `Close` button of any item is clicked.
+ *
+ * @param {HTMLElement} item the item about to be closed.
+ * @public
+ */
+@event<NotificationItemCloseEventDetail>("item-close", {
+	detail: {
+		/**
+		 * @public
+		 */
+		item: { type: HTMLElement },
+	},
+})
 
-	constructor() {
-		super();
-		this.accessibleName = NotificationList.i18nFioriBundle.getText(NOTIFICATION_LIST_ACCESSIBLE_NAME);
-		this._itemNavigation._navigationMode = NavigationMode.Auto;
+/**
+ * Fired when an item is toggled.
+ *
+ * @param {HTMLElement} item the toggled item.
+ * @public
+ */
+@event<NotificationItemToggleEventDetail>("item-toggle", {
+	detail: {
+		/**
+		 * @public
+		 */
+		item: { type: HTMLElement },
+	},
+})
+
+/**
+ * Fired when the user scrolls to the bottom of the list.
+ *
+ * **Note:** The event is fired when the `growing='Scroll'` property is enabled.
+ * @public
+ */
+@event("load-more")
```

Comment:
currently there is no "growing" property, so either delete this event and all the related logic, or add the property if it's needed